server/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```
@@ -93,7 +93,11 @@ public RegexpQueryBuilder(StreamInput in) throws IOException {
         super(in);
         fieldName = in.readString();
         value = in.readString();
-        syntaxFlagsValue = in.readVInt();
+        // IMPORTANT - older versions of Lucene/elasticsearch used 0xffff to
+        // represent ALL but newer versions of Lucene will error given syntax flag
+        // values > 0xff
+        // Here, we mute the bits beyond 0xff to provide BWC
+        syntaxFlagsValue = in.readVInt() & RegExp87.ALL;
```

only apply the mask if version is < 7.7, to not silently hide bugs?